caikit_nlp/modules/text_embedding/embedding.py

```diff
@@ -728,18 +785,19 @@ def _truncate_input_tokens(
                         f"maximum sequence length for this model ({tokens} > {max_tokens})."
                     ),
                 )
+        input_token_count = self._sum_token_count(tokenized)
 
-        return tokenized
+        return tokenized, input_token_count
 
     def encode(
         self,
-        sentences: Union[str, List[str]],
+        sentences: Union[str, Collection[str]],
         batch_size: int = 32,
-        device: str = None,
+        device: Optional[str] = None,
         convert_to_numpy: bool = True,
         convert_to_tensor: bool = False,
-        truncate_input_tokens: Optional[int] = 0,
-    ) -> np.ndarray:
+        truncate_input_tokens: int = 0,
+    ) -> EmbeddingResultTuple:
```

Comment:
I guess it is necessary change the return type because we need more info than the array can give us.

Reply:
My preference are to use named types for non-standard tuple returns. I find the development experience better.